hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/SerializationConstants.java

```diff
@@ -127,10 +127,12 @@
 
     public static final int JAVA_DEFAULT_TYPE_PRIORITY_QUEUE = -49;
 
+    public static final int JAVA_DEFAULT_TYPE_OPTIONAL = -50;
+
     // NUMBER OF CONSTANT SERIALIZERS...
-    public static final int CONSTANT_SERIALIZERS_LENGTH = 50;
+    public static final int CONSTANT_SERIALIZERS_LENGTH = 51;
 
-    public static final int JAVA_DEFAULT_TYPE_ENUM = -50;
+    public static final int JAVA_DEFAULT_TYPE_ENUM = -51;
```

Comment:
TYPE_ENUM seems to be unused, but for sure I just decreased indentifier.

Reply:
It could be used from our other projects like JET. Lets don't change it and add  `JAVA_DEFAULT_TYPE_OPTIONAL` as -51

Reply:
I have tried suggested change, but there is a small problem with indexes. `java.lang.ArrayIndexOutOfBoundsException: Index 51 out of bounds for length 51`

Each identifier is used as index to an array of registered serializers. ENUM is not registered and this causes null element on index 50 if `CONSTANT_SERIALIZERS_LENGTH` is set to 52.

Is the null acceptable in array of serializers? (At first glance it seems to me as some kind of access optimization that is trying to prevent null to exist in array)

Reply:
Hi @sgflt , I have double-checked. It seems that the enum serializer is actually deleted but accidentally put back when doing a merge. We can delete it completely.

Reply:
Hi @sancar. I've pushed commit with deletion of unused constant. Thanks for review.